Tools/CISupport/Shared/maybe-delete-directory.bash

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
```

Comment:
Nit: we'd normally use `.sh` extension for the file. the shebang looks fine.

Slightly more substantial, please use `/bin/sh`. The macOS version of `bash` is very old, and actually prints out warnings about pending removal if I remember correctly.

We'd probably use Python for something like this, but I think that shell is OK.

Reply:
`bash` on macOS prints a deprecation warning when invoked interactively but does not when invoked non-interactively (such as by `bash foo.bash`). This scripts uses a number of bash-isms that do not work in `sh`. This script does appear to work in `zsh` however. I've changed the shebang and file extension to `zsh`.

Reply:
No, please don't use zsh either. We standardized on /bin/sh shebang intentionally.

Reply:
Okay. I will rewrite this script to remove the bash-isms that `sh` does not support.